### ISSUE TITLE
advance has a misleading name

### ISSUE THREAD

**Issue by danielcompton** [juxt/tick#72]
`tick.core/advance` takes a Clock and a Duration. Based on the name and its proximity to clock related code, I assumed that this was for manually advancing a fixed clock. However, [Clock#tick](https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html#tick-java.time.Clock-java.time.Duration-) doesn't change the time on the clock, it just returns a new clock which ticks at a different resolution.

It could be good to add a docstring for this, rename it, or maybe remove it entirely?

**Comment by henryw374** [comment]
It sounds like you might be the first person to try using any atomic clock fns :(

I agree it should all be removed or looked over and tested properly.

**Comment by danielcompton** [comment]
I've found https://github.com/state-machine-systems/mock-clock which provides the mutability that I was after, while also being a Clock subclass. I think that is probably a good library to recommend to people if they want to mock clocks; it has no dependencies and is just a single file.

I had a go writing the same thing in Clojure (before I discovered mock-clock) and it was pretty tricky `proxy` code to extend the abstract base Clock class. It's doable to write this in Clojure but it's not pretty, and a mutable Java object is likely preferable.
